drivers/wifi/nrf_wifi/src/wpa_supp_if.c

```diff
@@ -1951,19 +1965,29 @@ void nrf_wifi_supp_event_proc_get_conn_info(void *if_priv,
 {
 	struct nrf_wifi_vif_ctx_zep *vif_ctx_zep = NULL;
 	struct wpa_conn_info *conn_info = NULL;
+	struct nrf_wifi_ctx_zep *rpu_ctx_zep = NULL;
+	unsigned int vif_ctx_cnt = 0;
 
 	if (!if_priv || !info) {
 		LOG_ERR("%s: Invalid params", __func__);
 		k_sem_give(&wait_for_event_sem);
 		return;
 	}
 	vif_ctx_zep = if_priv;
-	conn_info = vif_ctx_zep->conn_info;
-
-	conn_info->beacon_interval = info->beacon_interval;
-	conn_info->dtim_period = info->dtim_interval;
-	conn_info->twt_capable = info->twt_capable;
-	k_sem_give(&wait_for_event_sem);
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+	vif_ctx_cnt = nrf_wifi_fmac_get_num_vifs(rpu_ctx_zep->rpu_ctx);
+
+	/* Notify all VIFs */
+	for (int idx = 0; idx < vif_ctx_cnt; ++idx) {
+		vif_ctx_zep = nrf_wifi_get_vif_ctx_by_idx(idx);
+		if (vif_ctx_zep) {
+			conn_info = vif_ctx_zep->conn_info;
+			conn_info->beacon_interval = info->beacon_interval;
+			conn_info->dtim_period = info->dtim_interval;
+			conn_info->twt_capable = info->twt_capable;
+			k_sem_give(&wait_for_event_sem);
+		}
+	}
```

Comment:
same comment, this has to be done per-VIF, so, the current code should work fine.